app/src/main/java/com/example/android/architecture/blueprints/todoapp/di/ApplicationModule.kt

```diff
@@ -0,0 +1,66 @@
+package com.example.android.architecture.blueprints.todoapp.di
+
+import android.content.Context
+import androidx.room.Room
+import com.example.android.architecture.blueprints.todoapp.data.source.DefaultTasksRepository
+import com.example.android.architecture.blueprints.todoapp.data.source.TasksDataSource
+import com.example.android.architecture.blueprints.todoapp.data.source.TasksRepository
+import com.example.android.architecture.blueprints.todoapp.data.source.local.TasksLocalDataSource
+import com.example.android.architecture.blueprints.todoapp.data.source.local.ToDoDatabase
+import com.example.android.architecture.blueprints.todoapp.data.source.remote.TasksRemoteDataSource
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import javax.inject.Named
+import javax.inject.Singleton
+
+@Module
+class ApplicationModule {
+
+    @Singleton
+    @Provides
+    fun provideRepository(
+        @Named("TasksRemoteDataSource") tasksRemoteDataSource: TasksDataSource,
```

Comment:
I think doing annotations as qualifiers is much less error prone as you can then autocomplete them.
```
@Qualifier
@Retention(RUNTIME)
annotation class TasksRemoteDataSource
``` is fairly easy to declare at top of module file

Reply:
+1, I hate `@Named` and use it when I'm feeling super lazy. I should really move away from them.

Reply:
TIL. Done!